### ISSUE TITLE
Android NNAPI: Exception during initialization: unordered_map::at: key not found

### ISSUE THREAD

**Issue by ArtiX-GP** [microsoft/onnxruntime#11488]
Hi! I am install ONNX runtime lib use:
```
implementation 'com.microsoft.onnxruntime:onnxruntime-android:latest.release'
```
Because there ( https://onnxruntime.ai/docs/build/eps.html#nnapi ) says: 

> The pre-built ONNX Runtime Mobile package for Android includes the NNAPI EP.

In code I tried to use:

```
    public OrtSession createSession() throws IOException, OrtException {
        OrtSession.SessionOptions so = new OrtSession.SessionOptions();
        so.addNnapi();
//        It's doesn't work too.
//        EnumSet<NNAPIFlags> nnapiFlags = EnumSet.of(NNAPIFlags.USE_FP16);
//        so.addNnapi(nnapiFlags);
        return _Environment.createSession(getModel(), so); // getModel() returns byte array.
    }
```

But I am getting error

```
E/onnxruntime:  [E:onnxruntime:, inference_session.cc:1587 operator()] Exception during initialization: unordered_map::at: key not found
W/System.err: ai.onnxruntime.OrtException: Error code - ORT_RUNTIME_EXCEPTION - message: Exception during initialization: unordered_map::at: key not found
```

What I do wrong?

**System information**
- OS Platform and Distribution: Windows 10
- ONNX Runtime installed from (source or binary): using Gradle
- ONNX Runtime version: 1.11
- Android Device: Samsung Galaxy S9+

**Comment by Craigacp** [comment]
Does it work without the `addNnapi` call?

**Comment by edgchen1** [comment]
Would it be possible to share your model?
You can also try increasing the logging level so we can get more information.
E.g., pass ORT_LOGGING_LEVEL_VERBOSE to [getEnvironment](https://onnxruntime.ai/docs/api/java/ai/onnxruntime/OrtEnvironment.html#getEnvironment-ai.onnxruntime.OrtEnvironment.LoggingLevel-).

**Comment by ArtiX-GP** [comment]
Hi, @Craigacp @edgchen1
Yes, without addNnapi call it works.

I published models, you can download it: https://disk.yandex.ru/d/dyGIz15cj0iFyQ

**Comment by ArtiX-GP** [comment]
I will write about the results of logging (ORT_LOGGING_LEVEL_VERBOSE) a little later

**Comment by ArtiX-GP** [comment]
Hi, @edgchen1 !
Sorry for delay. I published log
[ONNXRuntime_Log_android.txt](https://github.com/microsoft/onnxruntime/files/8693108/ONNXRuntime_Log_android.txt)

Environment created:
`_Environment = OrtEnvironment.getEnvironment(OrtLoggingLevel.ORT_LOGGING_LEVEL_VERBOSE);`

MainActivity: 328 line
`return _Environment.createSession(getModel(), so); // getModel() returns byte array.`

**Comment by edgchen1** [comment]
Thanks for posting the log. I am also able to reproduce the issue on an Android device. Looking into it.

**Comment by edgchen1** [comment]
Verified that the models can load and run in master. This issue should have been fixed by https://github.com/microsoft/onnxruntime/commit/e3ff4a6bfaf1f8a739317fae006f736bd487bacc.

You could try just using the CPU EP. The performance may not be very good with NNAPI, as there are unsupported nodes that cause numerous partitions. From the log:
```
[W:onnxruntime:ort-java, nnapi_execution_provider.cc:197 GetCapability] NnapiExecutionProvider::GetCapability, number of partitions supported by NNAPI: 42 number of nodes in the graph: 900 number of nodes supported by NNAPI: 759
```

Or you could try building from source.
https://onnxruntime.ai/docs/build/android.html
https://onnxruntime.ai/docs/build/custom.html#android

**Comment by skottmckay** [comment]
Fixed in 1.12 release
